### DISCUSSION TITLE
pfSense -> Snort -> Syslog -> SO/Elasticsearch

### DISCUSSION THREAD

**Discussion by arisentow** [Security-Onion-Solutions/securityonion#8963] [Unsupported Versions]
Hello all.

I've configured pfSense to send logs to Security Onion via syslog, including Snort alerts. I can see the Snort alerts in Kibana, but I am looking for a way to extract/parse the fields from the message in both Kibana/Elasticsearch and the Security Onion dashboard. 

I'm assuming that it can be done via Ingest Pipelines, however I've never developed pipelines in Elasticsearch and am having a difficult time trying to figure it out. I have done this successfully with pipelines in GrayLog through regex, but it doesn't look like those pipelines translate to Elasticsearch.

Any help would be greatly appreciated. I was hoping there would be some "off the shelf" processors out there, but I haven't been able to find any.

**Comment by dougburks** [comment]
Most folks just have Security Onion monitor the traffic hitting their pfSense firewall so that Security Onion automatically generates its own NIDS alerts without having to worry about having to parse NIDS alerts from elsewhere. This strategy also has the added bonus of recording not just NIDS alerts but also protocol metadata and full packet capture. In addition, you can still configure pfSense to send its actual firewall logs to Security Onion and they will get parsed automatically allowing you to correlate all these logs together:
https://docs.securityonion.net/en/2.3/other-supported-logs.html#example-pfsense

**Reply by arisentow** [reply]
Doug, thank you for the response. I actually sort of revised my question in this post: #8975. The problem that I'm having is that the pfSense logs are not being parsed.

Part of the reason for wanting to parse out the logs from pfSense is that I currently have Snort running there with alerts being sent to the pfSense logs, while running Suricata on SO. From what I understand I can't run both (correct me if I'm wrong) within SO. Suricata and Snort trigger different alerts, so looking to consolidate within SO. 

Thanks!

**Reply by dougburks** [reply]
> Suricata and Snort trigger different alerts

That's not necessarily the case. If you run the Emerging Threats ruleset on both Suricata and Snort, then both of them should (in theory) generate very similar alerts. 

So the real question is: what ruleset are you running on your pfSense firewall and why?

**Reply by arisentow** [reply]
Doug,

I'm also running the Snort Subscriber and GPLv2 Community Rules

**Reply by dougburks** [reply]
You can run the Snort Subscriber and GPLv2 Community rules on Suricata. The only caveat is that some rules are Shared Object rules and those will not load in Suricata. For more information, please see https://docs.securityonion.net/en/2.3/rules.html.

**Reply by arisentow** [reply]
![image](https://user-images.githubusercontent.com/116360047/198140895-3adbc0a6-d212-4951-8406-4e616174cceb.png)



Looks like it might be working with this. I know it at least fetched the rules after doing a 'so-rule-update', not sure if they were applied. The output says 'added: 0'...

![image](https://user-images.githubusercontent.com/116360047/198140459-0017e223-62ef-4096-afbe-999e4ab54580.png)
